content/v2.0/router/basic-usage.md

```diff
@@ -0,0 +1,221 @@
+get '/action',     to: "home.index"
+get '/middleware', to: Middleware
+get '/rack-app',   to: RackApp.new
+get '/rails',      to: ActionControllerSubclass.action(:new)
+```
+
+When we use a string, it tries to instantiate a class from it:
+
+```ruby
+get '/rack-app', to: 'rack_app' # it will map to RackApp.new
+```
+
+## Actions
+
+Full Rack integration is great, but the most common endpoint that we'll use in our web applications is an **action**.
+Actions are objects responsible for responding to incoming HTTP requests.
+They have a nested naming like `Main::Actions::Home::Index`.
+This is a really long name to write, that's why Hanami has a **naming convention** for it: `"home.index"`.
+
+```ruby
+# /config/routes.rb
+
+slice :main, at: '/' do
+  root to: "home.index" # => will route to Web::Controllers::Home::Index
+end
+```
+
+The first token is the name of the controller `"home"` is translated to `Home`.
+The same transformation will be applied to the name after the `.`: `"index"` to `Index`.
+
+Hanami can figure out the namespace (`Main::Actions`) based on the slice the route belongs to, and compose the full class name.
+
+```ruby
+# /slices/main/lib/main/actions/home/index.rb
+
+module Main
+  module Actions
+    module Home
+      class Index < Hanami::Action
+      end
+    end
+  end
+end
+```
+
+`Home::Index` action of the `Main` slice, will call a corresponding view object with prepared parameters. The view then will render the prepared `slim` template.
+
+```ruby
+# /slices/main/lib/main/views/home/index.rb
+
+module Main
+  module Views
+    module Home
+      class Index < Main::View
+      end
+    end
+  end
+end
+```
+
+```ruby
+# /slices/main/lib/main/templates/home/index.html.slim
+
+h1 Hello, Hanami!
+```
+
+This is great, as it allows us to write our templates in the `HTML` or `Slim` files, or serialize our `JSON` responses using serializers instead of using raw strings everywhere!
+
+Read more about [Hanami::Actions](/v2.0/actions/overview) and [rendering views](/v2.0/views/overview)!.
+
+# Features
+
+`Hanami::Router` comes with a full set of features helping to organize your routes, no matter how big or complex your application is. It's designed to scale!
+
+## Slices
+
+One of the unique Hanami features is how it encourages modularizing your application into multiple, self-contained building blocks. We call it [slices](/v2.0/slices/overview).
+
+A slice is an independent piece of application. Think of it as different entry points to your domain.
+
+```ruby
+Hanami.application.routes do
+  slice :main, at: "/" do
+    root to: "home.show"  # slices/main/lib/actions/home/show.rb
+    get '/hello', to: 'hello.index' # slices/main/lib/actions/hello/index.rb
+  end
+  
+  slice :admin, at: "/" do
+    root to: "dashboard.show" # slices/admin/lib/actions/dashboard/show.rb
+    get '/users', to: 'users.index' # slices/admin/lib/actions/users/index.rb
+  end
+end
+```
+
+Read more about [slices](/v2.0/slices/overview)!
```

Comment:
@timriley is it more to this to include here? Or is it fine to put the reset in the `slices` section` ?

Reply:
I'm not Tim but I think this is a good approach. There's a `slice` method in the router, so that's what's relevant, but any more info (about how slices themselves work) should go in the slices documentation :)